tomviz/RotateAlignWidget.cxx

```diff
@@ -509,9 +510,29 @@ void RotateAlignWidget::onFinalReconButtonPressed()
   t = vtkTrivialProducer::SafeDownCast(output->producer()->GetClientSideObject());
   vtkImageData *recon = vtkImageData::SafeDownCast(t->GetOutputDataObject(0));
 
-
   LoadDataReaction::dataSourceAdded(output);
+ 
   */
+  //Apply python transform
+  //Apply shift (in y-direction) first
+  QMap<QString, QString> substitutions;
+  substitutions.insert("###SHIFT###",
+                       QString("SHIFT = [%1, %2, %3]").arg(0)
+                       .arg(-this->Internals->Ui.rotationAxis->value()).arg(0));
+
+  QString scriptLabel = "Shift";
+  QString scriptSource = readInPythonScript("Shift3D");
+  AddPythonTransformReaction::addPythonOperator(this->Internals->Source, scriptLabel, scriptSource, substitutions);
+  
```

Comment:
You may want to add
```c++
substitutions.clear();
```
call here.  It doesn't really matter unless `###SHIFT###` appears in your second script though.